src/optimizer/projection_push_down_optimizer.cpp

```diff
@@ -214,8 +214,18 @@ binder::expression_vector ProjectionPushDownOptimizer::pruneExpressions(
     const binder::expression_vector& expressions) {
     expression_set expressionsAfterPruning;
     for (auto& expression : expressions) {
-        if (expression->expressionType != common::PROPERTY ||
-            propertiesInUse.contains(expression)) {
+        switch (expression->expressionType) {
+        case common::VARIABLE: {
```

Comment:
Can you explain why this had to change and what exactly is happening here? Maybe add some comments too.